tests/system/apmserver.py

```diff
@@ -155,26 +158,10 @@ def dump_diagnotics(self, interval=2):
                     out.write("failed to query tasks: {}\n".format(e))
 
     def wait_until(self, cond, max_timeout=10, poll_interval=0.25, name="cond"):
-        """
-        Like beat.beat.wait_until but catches exceptions
-        In a ElasticTest `cond` will usually be a query, and we need to keep retrying
-         eg. on 503 response codes
-        """
-        assert callable(cond), "First argument of wait_until must be a function"
-
-        start = datetime.now()
-        while datetime.now()-start < timedelta(seconds=max_timeout):
-            try:
-                result = cond()
-                if result:
-                    return result
-            except AttributeError as ex:
-                raise ex
-            except:
-                pass
-            time.sleep(poll_interval)
-        raise TimeoutError("Timeout waiting for '{}' to be true. ".format(name) +
-                           "Waited {} seconds.".format(max_timeout))
+        wait_until(cond, max_timeout=max_timeout, poll_interval=poll_interval, name=name)
+
+# If you implement a Subclass of ServerSetUpBaseTest ensure to implement a `tearDown` function taking care of
```

Comment:
We could also use https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup, which gets called regardless of whether `setUp` succeeds

Reply:
Working on such an alternative at the moment.